### PR TITLE
feat(sortByPath): sort list according to paths

### PR DESCRIPTION
**Pull request by nolleto**
Implemented `sortByPath` function.

Issue: #1240

### CODE REVIEW

File: src/sortByPath.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * ], people) // [bob, alice, clara]
+ * sortByProps([
+ *   ['address', 'zipCode'],
+ *   ['address', 'street'],
+ * ], people) // [bob, clara, alice]
+ */
+
+const sortByPath = useWith(sortWith, [mapPathsToAscendSort, identity]);
+
+export default sortByPath;
```
Comment (nolleto) [inline src/sortByPath.js:54]: I was wondering if the shouldn't be better to name it as `sortByPaths` (with `s` in the end) because it accepts multiple paths as `sortByProps` accept multiple props. 🤔 

cc @char0n

File: src/sortByPath.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Sort a list of objects by a list of paths (if first prop value is equivalent, sort by second, etc).
+ *
+ * @func sortByPath
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.29.0|v2.29.0}
+ * @category List
+ * @sig [[k]] -> [{k: v}] -> [{k: v}]
+ * @param {Array.<string>} paths A list of paths in the list param to sort by
```
Comment (nolleto) [inline src/sortByPath.js:14]: Maybe `Array.<object>`?

File: src/sortByPath.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+const pathToAscendSort = pipe(path, ascend);
+const mapPathsToAscendSort = map(pathToAscendSort);
+
+/**
+ * Sort a list of objects by a list of paths (if first prop value is equivalent, sort by second, etc).
+ *
+ * @func sortByPath
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.29.0|v2.29.0}
```
Comment (nolleto) [inline src/sortByPath.js:11]: Not sure about the version

File: src/sortByPath.js (hunk 1 above)
Comment (char0n) [reply src/sortByPath.js:54]: @nolleto yes, `sortByPaths` is the name we want to go for, so that it's compatible with `RA.sortByProps`.

File: src/sortByPath.js (hunk 2 above)
Comment (char0n) [reply src/sortByPath.js:14]: The `paths` param is a list of list of strings: `Array.<Array.<string>>`

File: src/sortByPath.js (hunk 3 above)
Comment (char0n) [reply src/sortByPath.js:11]: Don't worry about it,  I'll change it before the merge.

File: src/sortByPath.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { ascend, identity, map, path, pipe, sortWith, useWith } from 'ramda';
+
+const pathToAscendSort = pipe(path, ascend);
+const mapPathsToAscendSort = map(pathToAscendSort);
+
+/**
+ * Sort a list of objects by a list of paths (if first prop value is equivalent, sort by second, etc).
```
Comment (char0n) [inline src/sortByPath.js:7]:
```suggestion
 * Sort a list of objects by a list of paths (if first path value is equivalent, sort by second, etc).
```

File: test/sortByPath.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          address: {
+            street: 32,
+          },
+        };
+        const people = [clara, bob, alice];
+
+        assert.deepEqual(sortByAddress(people), [bob, alice, clara]);
+      });
+    });
+  });
```
Comment (char0n) [inline test/sortByPath.js:84]: Adding a test case for providing empty list of paths would be good to: `[]`

**Review by char0n — APPROVED** [review]
Hi, thank you for contributing and sorry that it took me some time to respond. Let's keep this PR moving if you're still interested in contributing. Your submission will be used by thousands of projects worldwide ;]

**Comment by char0n** [comment]
Also please rebase this PR on top of current master.

**Comment by nolleto** [comment]
@char0n I just updated the branch according to your code review 😃

**Review by char0n — APPROVED** [review]
Thanks for contributing! I'll do some additional changes  before the merge.

**Comment by char0n** [comment]
Additional changes before I can merge:

I need you to add

**Comment by char0n** [comment]
`

**Comment by char0n** [comment]
Additional changes I need you to make before the merge:

Add following code in `types/index.d.ts` on line 1486 right after `sortByProps`.

```ts
        /**
         * Sort a list of objects by a list of paths (if first path value is equivalent, sort by second, etc).
         */
        sortByPaths(props: string[][], list: object[]): object[];
        sortByPaths(props: string[][]): (list: object[]) => object[];
```

Create new file called `sortByPaths.ts`  in `types/tests/` and put following code in it:

```ts
import * as RA from 'ramda-adjunct';

const alice = {
    name: 'Alice',
    address: {
        street: 31,
        zipCode: 97777,
    },
};
const bob = {
    name: 'Bob',
    address: {
        street: 31,
        zipCode: 55555,
    },
};
const clara = {
    name: 'Clara',
    address: {
        street: 32,
        zipCode: 90210,
    },
};
const people = [clara, bob, alice];

RA.sortByPaths([['name']])(people); // $ExpectType object[]
RA.sortByPaths([['address', 'street'], ['address', 'zipCode']])(people); // $ExpectType object[]
RA.sortByPaths([])(people); // $ExpectType object[]

RA.sortByPaths([['name']])(123); // $ExpectError
RA.sortByPaths(123)(people); // $ExpectError
```

**Review by char0n — APPROVED** [review]
Merging. THanks again!
